src/Features/CSharp/Portable/AddDebuggerDisplay/CSharpAddDebuggerDisplayCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.AddDebuggerDisplay;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.AddDebuggerDisplay
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(CSharpAddDebuggerDisplayCodeRefactoringProvider)), Shared]
+    internal sealed class CSharpAddDebuggerDisplayCodeRefactoringProvider : AbstractAddDebuggerDisplayCodeRefactoringProvider<TypeDeclarationSyntax, MethodDeclarationSyntax>
+    {
+        protected override bool HasDebuggerDisplayAttribute(TypeDeclarationSyntax typeDeclaration)
+        {
+            return typeDeclaration.AttributeLists
+                .SelectMany(list => list.Attributes)
+                .Any(IsDebuggerDisplayAttribute);
+        }
```

Comment:
this is being done syntactically, but it could easily (and cheaply) be done semantically.  That will work for both C# and VB.  That would be my strong recommendation unless there is super compelling data indicating we should do it this way. :)

Reply:
Whichever makes the tests pass and is less complex.

Reply:
semantics then :)